src/main/java/com/thebois/models/inventory/Inventory.java

```diff
@@ -12,16 +12,56 @@
 public class Inventory implements IInventory {
 
     private final ArrayList<IItem> items = new ArrayList<>();
+    private final float maxCapacity;
+
+    /**
+     * Instantiates with infinite capacity.
+     */
+    public Inventory() {
+        this.maxCapacity = Float.MIN_VALUE;
+    }
+
+    /**
+     * Instantiates with a given maximum carrying capacity.
+     *
+     * @param maxCapacity The maximum weight, in kilograms, that can be contained.
+     */
+    public Inventory(final float maxCapacity) {
+        this.maxCapacity = maxCapacity;
+    }
+
+    @Override
+    public boolean isFull() {
+        final float weight = calculateTotalWeight();
+        return atMaxCapacity(weight);
+    }
+
+    private float calculateTotalWeight() {
+        return items.stream().map(IItem::getWeight).reduce(0f, Float::sum);
+    }
+
+    private boolean atMaxCapacity(final float weight) {
+        if (hasInfiniteCapacity()) return false;
+        return weight >= maxCapacity;
+    }
+
+    private boolean hasInfiniteCapacity() {
+        return maxCapacity == Float.MIN_VALUE;
+    }
 
     @Override
-    public void add(final IItem item) {
+    public boolean tryAdd(final IItem item) {
```

Comment:
Question (sort of unrelated)

But when should methods return a boolean over void? Any rules for it? Or is it more like "this makes it easier for us?" Because this makes testing easier :P

Reply:
>But when should methods return a boolean over void?

When there is a result to return.

>Any rules for it?

Yeah, if there is a meaningful result that can be returned. Methods that follow the pattern `tryDoSomething(arguments)` should always return in some way if they succeeded or failed. Otherwise they should just be called `doSomething`. The fact that it's a `try` method means that it can fail. The client definitely would want to handle this case, and so it needs to be able to tell if the operation failed or succeeded, which is what the return value indicates. This allows for some really elegant code:
```java
if (tryAdd(item)) {
    // Item was added, continue logic here
}
```
```java
if (!tryAdd(item)) {
    // Item could not be added, notify the user
}
```
What's really nice is that the client can also make the choice to ignore the returned value and handle the method as if it was of type `void`:
```java
tryAdd(item);
// I don't care if the item has been added or not, let's just continue.
```
This would not be possible if a checked exception was used instead.

>Because this makes testing easier :P

Does it? The tests I've written had to be made exactly to ensure that this functionality worked, so it didn't make it easier, it was actually harder since I had to implement more tests with more complex test cases. Having it return void would be easier to test.